### PR TITLE
feat: pdp-structure

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

**Comment by kibo-chandradeeptalaha** [comment]
Leaving the integration test for now. Will work on it after integration of the component with custom hooks.

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+            Product Specs
+          </Typography>
+        }
+      />
+      <Divider />
+      <Collapse in={expanded} timeout="auto" unmountOnExit>
+        <CardContent>
+          {props?.properties.map((option: ProductProperties, index: number) => {
+            return (
+              <Box data-testid="productOption" key={index}>
```
Comment (kibo-kevinwatts) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:327]: dont use index as key

File: lib/types/Breadcrumb.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export type BreadCrumb = Record<string, string | null | undefined>
+export type BreadcrumbsListReturnType = BreadCrumb[]
```
Comment (kibo-kevinwatts) [inline lib/types/Breadcrumb.ts:2]: I don't think we need to create a new type here, just use `BreadCrumb[]`

File: next.config.js [hunk 3]
```diff
@@ -11,6 +11,11 @@ module.exports = {
       'cdn-tp3.mozu.com',
       'cdn-tp4.mozu.com',
       'cdn-sb.mozu.com',
+      'imgs.search.brave.com',
```
Comment (kibo-kevinwatts) [inline next.config.js:14]: remove this domain and use images on our cdn

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+import type {
+  AttributeDetail,
+  ProductImage,
+  ProductOption,
+  ProductOptionValue,
+} from '@/lib/gql/types'
+
+interface ProductDetailTemplateProps {
+  product: ProductCustom
```
Comment (kibo-kevinwatts) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:44]: lets add "breadcrumbs" as a prop, to be consistent with the PLP

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+import KiboBreadcrumbs from '@/components/core/Breadcrumbs/KiboBreadcrumbs'
+import ImageGallery from '@/components/core/ImageGallery/ImageGallery'
+import ColorSelector from '@/components/product/ColorSelector/ColorSelector'
+import ProductOptionCheckbox from '@/components/product/ProductOptionCheckbox/ProductOptionCheckbox'
+import ProductOptionSelect from '@/components/product/ProductOptionSelect/ProductOptionSelect'
+import ProductOptionTextBox from '@/components/product/ProductOptionTextBox/ProductOptionTextBox'
+import ProductRecommendations from '@/components/product/ProductRecommendations/ProductRecommendations'
+import ProductVariantSizeSelector from '@/components/product/ProductVariantSizeSelector/ProductVariantSizeSelector'
+import { productGetters } from '@/lib/getters'
+import type { ProductCustom } from '@/lib/types'
```
Comment (kibo-kevinwatts) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:33]: import type { ProductProperties, ProductCustom } on same line

**Review by kibo-kevinwatts — APPROVED** [review]
very minor quick changes, otherwise looks good!

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 4 above)
Comment (kibo-chandradeeptalaha) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx:44]: I actually fixed this in the icky-609 PR

File: tsconfig.json [hunk 6]
```diff
@@ -22,6 +22,11 @@
       "@/types/*": ["types/*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    "lib/helpers/validateProductVariations.spec.js"
```
[the remark this answers is not in the document]
Comment (kibo-chandradeeptalaha) [reply tsconfig.json:29]: Done

File: test-report.xml [hunk 7]
```diff
@@ -1,208 +1,282 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (kibo-chandradeeptalaha) [reply test-report.xml:1]: Done

**Pull request merged**